### ISSUE TITLE
Test platform should be published to a separate directory than dotnet.dll

### ISSUE THREAD

**Issue by livarcocc** [Microsoft/vstest#816]
_From @eerhardt on January 26, 2017 5:11_

The `microsoft.testplatform.cli` nuget package is getting published to the sdk folder along with the rest of the CLI assemblies.  However, this nuget package ships a bunch of assemblies (like `System.ComponentModel.Primitives.dll`) in its contentFiles directory.  When the CLI wants to move to different versions of the framework, this causes problems because we have these previous versions (think .NET Core 1.0) in our sdk folder.  Normally this isn't a problem, but when we cross-gen, cross-gen sees these assemblies and fails because they are the wrong version.  It is looking for .NET Core 2.0 versions.

To fix these types of issues, we should isolate the testplatform into its own folder, like we do with Roslyn.

_Copied from original issue: dotnet/cli#5464_

**Comment by livarcocc** [comment]
_From @piotrpMSFT on February 1, 2017 9:19_

Agreed. The test platform already made targets changes to enable this scenario. The change is a bit too risky for 1.0 but we plan to undertake it in 2.0.

**Comment by livarcocc** [comment]
_From @eerhardt on May 15, 2017 16:59_

Re-pinging this issue as this continues to cause issues.

Not only should the test platform be published to a separate directory, but we need to restructure the NuGet package that the vstest team gives to the SDK.

The issue here is that vstest "publishes" against version XYZ of the shared fx, and says "here are the 15 assemblies we need to put into the SDK".  And then later the CLI picks up version ABC of the shared fx, which is different than the version vstest used. And the CLI tries taking the vstest assemblies and put them on the new version of the shared fx.  However, the new version of the shared fx may not have the same set of assemblies in it as what vstest used to publish.  Thus the CLI would be blocked on taking the new shared fx.  The CLI would need a new vstest drop that used ABC of the shared fx before it could consume the new shared fx.


Here is a concrete example of this issue happening: https://github.com/dotnet/cli/pull/6582#issuecomment-301530088.

@livarcocc - Do we need to open a new issue on the vstest team to get this restructuring to happen?  Can you please drive this restructuring so we stop running into these issues?

/cc @codito @Faizan2304

**Comment by livarcocc** [comment]
Yeah, we will chat about it tonight and then I will move this bug to the vstest repo.

**Comment by eerhardt** [comment]
The title of this should probably change to something like "Test platform should use typical NuGet packages to distribute assemblies into .NET Core SDK".

The main driving concern here is that we aren't getting dependency unification and any other tooling support because all of the test platform assemblies are shipped as "contentFiles".

**Comment by ShreyasRmsft** [comment]
Have added it to [back log](https://mseng.visualstudio.com/VSOnline/_workitems/edit/1225270). Closing this issue.
